ddtrace/util.py

```diff
@@ -1,137 +1,20 @@
-      If is isn't, nothing to worry about, patch the key as usual
-      But if it is, search for a "__dd_orig_{key}" method on the class, which is
-      the original unpatched method we wish to trace.
-
-    """
-    def _get_original_method(thing, key):
-        orig = None
-        if hasattr(thing, '_dogtraced'):
-            # Search for original method
-            orig = getattr(thing, "__dd_orig_{}".format(key), None)
-        else:
-            orig = getattr(thing, key)
-            # Set it for the next time we attempt to patch `thing`
-            setattr(thing, "__dd_orig_{}".format(key), orig)
-
-        return orig
-
-    if inspect.isclass(patchable) or inspect.ismodule(patchable):
-        orig = _get_original_method(patchable, key)
-        if not orig:
-            # Should never happen
-            return
-    elif hasattr(patchable, '__class__'):
-        orig = _get_original_method(patchable.__class__, key)
-        if not orig:
-            # Should never happen
-            return
-    else:
-        return
-
-    dest = patch_func(orig, service, meta, tracer)
-
-    if inspect.isclass(patchable) or inspect.ismodule(patchable):
-        setattr(patchable, key, dest)
-    elif hasattr(patchable, '__class__'):
-        setattr(patchable, key, dest.__get__(patchable, patchable.__class__))
-
-
-def asbool(value):
-    """Convert the given String to a boolean object. Accepted
-    values are `True` and `1`."""
-    if value is None:
-        return False
-
-    if isinstance(value, bool):
-        return value
-
-    return value.lower() in ("true", "1")
-
-
-def get_env(integration, variable, default=None):
-    """Retrieves environment variables value for the given integration. It must be used
-    for consistency between integrations. The implementation is backward compatible
-    with legacy nomenclature:
-        * `DATADOG_` is a legacy prefix with lower priority
-        * `DD_` environment variables have the highest priority
-        * the environment variable is built concatenating `integration` and `variable`
-          arguments
-        * return `default` otherwise
-    """
-    key = '{}_{}'.format(integration, variable).upper()
-    legacy_env = 'DATADOG_{}'.format(key)
-    env = 'DD_{}'.format(key)
-
-    # [Backward compatibility]: `DATADOG_` variables should be supported;
-    # add a deprecation warning later if it's used, so that we can drop the key
-    # in newer releases.
-    value = os.getenv(env) or os.getenv(legacy_env)
-    return value if value else default
-
-
-def unwrap(obj, attr):
-    f = getattr(obj, attr, None)
-    if f and isinstance(f, wrapt.ObjectProxy) and hasattr(f, '__wrapped__'):
-        setattr(obj, attr, f.__wrapped__)
+# [Backward compatibility]: keep importing modules functions
+from .utils.deprecation import deprecated, deprecation
+from .utils.formats import asbool, deep_getattr, get_env
+from .utils.wrappers import safe_patch, unwrap
+
+
+deprecation(
```

Comment:
If deprecation warnings are activated, this action will trigger a deprecation message (and will not crash with an import fail):
```python
from ddtrace.util import asbool
```